baremaps-core/src/main/java/org/apache/baremaps/iploc/data/Ipv4.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.baremaps.iploc.data;
+
+
+
+import com.google.common.primitives.Bytes;
+import java.math.BigInteger;
+import java.util.Arrays;
+import net.ripe.ipresource.Ipv4Address;
+import net.ripe.ipresource.UniqueIpResource;
+
+/**
+ * Represents an IPv4
+ */
+public class Ipv4 {
```

Comment:
Overall, it looks good to me, but couldn't we use the ipresource objects directly without introducing our Ipv4 and Ipv6 objects?

Reply:
I believe we should make an abstraction on top of `IpResource` in case we decide to use a different library later. Ipv4 will "wrap" `IpResource` and hide the library below.
I have changed the constructor from `IpResource` to protected in order to hide it.